src-java/server42/server42-stats/src/main/java/org/openkilda/server42/stats/zeromq/StatsCollector.java

```diff
@@ -73,9 +73,13 @@ public void run() {
         sessionId = RandomStringUtils.randomAlphanumeric(8);
         log.info("started with session id {}", sessionId);
         while (!isInterrupted()) {
-            try (Socket server = context.createSocket(ZMQ.PULL)) {
+
+            Socket serverRef = null;
+            try {
+                Socket server = context.createSocket(ZMQ.PULL);
                 server.setReceiveTimeOut(1000);
                 server.connect(connectEndpoint);
+                serverRef = server;
```

Comment:
Can/should it be done on line `Socket server = context.createSocket(ZMQ.PULL);`? I suspect that "some" resource allocation happens inside `.createSocket(...)` call, not only inside `.connect(...)` call.

Reply:
done